src/addrman.h

```diff
@@ -608,23 +625,19 @@ friend class CAddrManTest;
     void Check()
```

Comment:
I was expecting to see `EXCLUSIVE_LOCKS_REQUIRED(!cs)` added to these functions that lock `cs` to get compile time detection of double-locking. Any reason not to have them?

Reply:
Of course! Done in 7f01942d2d3d9d66a8932d6a1fe08281d9b52273.